src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,109 @@
-import React from 'react';
+/* eslint-disable react/button-has-type */
+import React, { FormEvent, useState } from 'react';
+import cn from 'classnames';
+import { Store } from '../store';
+import { client } from '../utils/fetchClient';
 
 export const NewCommentForm: React.FC = () => {
+  const {
+    selectedPost,
+    setPostComments,
+    setPostCommentsError,
+    setPostCommentsLoading,
+  } = Store();
+
+  const [authorName, setAuthorName] = useState('');
+  const [isNameEmpty, setIsNameEmpty] = useState(false);
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [isEmailEmpty, setIsEmailEmpty] = useState(false);
+  const [comment, setComment] = useState('');
+  const [isCommentEmpty, setIsCommentEmpty] = useState(false);
```

Comment:
we combine it in one comment state(the same with errros)

useState({
  author: '',
  email: '',
  comment: ''
})

Reply:
You might not clear understand how it works. If one state get update - whole Component function calls again. So each variable in function body will creates again

Reply:
> You might not clear understand how it works. If one state get update - whole Component function calls again. So each variable in function body will creates again. And rerender whole component, not some part of it

read about that a few minutes ago & deleted the comment)))